### DISCUSSION TITLE
[Feature] Show face images for a person

### DISCUSSION THREAD

**Discussion by akropp** [immich-app/immich#5558] [Feature Request]
### The feature

When looking at a person's page, you can't tell if faces are properly labelled when there are multiple people in a picture without clicking through to each picture and looking at the info tab.  If there could be a "show faces" toggle where each image is replaced with just the face that is tagged with the selected person, it would make it easy to identify mis-labelled faces.

The page would just look like the "People" page, except it would be the one matched face per image corresponding to the person you selected.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by SVNKoch** [comment]
I like that idea. How would you handle pictures where a person is taged multiple times? For example in a selfie with a drawing of one self or visible twice due to mirrors

**Reply by hoffleit** [reply]
If a person is tagged twice on a picture, the person's page should show two faces for this picture when "show faces" is active.

With "Show faces" the person's page would show all faces of the person.
Without "show faces", it would show pictures containing the person.

I completely agree with this feature request. I would even suggest to make "show faces" the default of the person's page.

**Comment by DrD4ffy** [comment]
I am currently still using Synology Photos. When correcting the assignment, you can only see the face there. In many cases this is enough, but more often than I would like, I wish I had the complete picture. The face is then perhaps too small, too blurred or people resemble each other. But with the context that the complete picture gives me, I know who it is. However, I am also familiar with the "which face in the photo is it" case described earlier in this thread. A combination of both views seems to be the most helpful for checking.

**Comment by shumancw** [comment]
My input:
Issue: Especially in family photos where there can be a lot of similarities between people, there are many time when the photos presented contain two people that look similar and the model chooses the wrong person.

Example: 10 photos are presented containing a group of people. The person I think is the person identified in all 10 photos is actually only identified correctly in 8 of the photos. In 2 of the photos presented it actually identified a cousin, for example, as being the same as the person in the other 8 photos.

Feature: Give the ability for the 10 pictures of the unnamed person to show as thumbnails or facebox in version of the person identified. Doing this allow for a user to quickly confirm that 8 of the photos belong to one person and 2 to another.

Could bundle with #6013

**Reply by sksbir** [reply]
Hi
This feature currently exists on Librephotos, which have many drawbacks comparing to Immich, but at least this advantage...
Really convenient to detect quickly bad faces associations.

**Comment by ianfixes** [comment]
It's worth pointing out that this capability already exists, because it's used in the show/hide person page.

I filed #27276 not realizing that this request already existed -- this title isn't very descriptive (don't all photos of people have a face??).

Could the title be updated to include a word like "crop" or "zoom" or "bounding box" for better clarity / searchability?

**Comment by badbob001** [comment]
Google's old Picasa app had this feature. Select a person and it'll show all photos with that person. In the corner are buttons to toggle the view to show the whole photo or just the cropped portion of the person.
<img width="844" height="675" alt="image" src="https://github.com/user-attachments/assets/97a86caa-4e32-4fd3-9caa-706b5ce32b99" />
I just started with immich and I'm finding it very difficult to cleanup the people-assignments without having the person-only view. Are the cropped people images already pre-generated or are they generated on-demand when updating a person's feature photo?
